src/main/java/com/app/store/rest/BookController.java

```diff
@@ -41,9 +47,9 @@ public BookController(BookService bookService) {
    * @return the book
    */
   @GetMapping("/{id}")
-  public ResponseEntity<Book> getById(@PathVariable UUID id) {
-    Book book = bookService.findById(id);
-    return ResponseEntity.ok(book);
+  public ResponseEntity<Optional<Book>> getById(@PathVariable UUID id) {
+    Optional<Book> book = Optional.ofNullable(bookService.findById(id));
+    return book.map(value -> new ResponseEntity(value , HttpStatus.OK)).orElseGet(() -> ResponseEntity.notFound().build());
```

Comment:
Good Job,
There is another solution to handle the notfound case, 
since the booksevice throws BookNotFoundException
we can manage the return of the controller by creating an exception handler dedicated class.

Reply:
@ControllerAdvice
public class RestExceptionHandler {

    @ExceptionHandler(BookNotFoundException.class)
    @ResponseStatus(value = HttpStatus.NOT_FOUND)
    public void handleBookNotFoundException() {
         } 
}